rocketpy/rocket/rocket.py

```diff
@@ -1564,13 +1691,22 @@ def set_rail_buttons(
         rail_buttons : RailButtons
             RailButtons object created
         """
+        radius = radius or self.radius
         buttons_distance = abs(upper_button_position - lower_button_position)
         rail_buttons = RailButtons(
-            buttons_distance=buttons_distance, angular_position=angular_position
+            buttons_distance=buttons_distance,
+            angular_position=angular_position,
+            rocket_radius=radius,
         )
         self.rail_buttons = Components()
-        rail_buttons.rocket_radius = radius or self.radius
-        self.rail_buttons.add(rail_buttons, lower_button_position)
+        position = Vector(
+            [
+                radius * -math.sin(math.radians(angular_position)),
+                radius * math.cos(math.radians(angular_position)),
+                lower_button_position,
+            ]
+        )
+        self.rail_buttons.add(rail_buttons, position)
         return rail_buttons
 
     def add_cm_eccentricity(self, x, y):
```

Comment:
has this method always existed? 

```suggestion
    def add_cm_eccentricity(self, x, y):
```

Reply:
It is just a simpler way to add x and y coordinates to the center of mass. This simpler way makes more sense with the newly defined coordinate systems